docs/hooks.md

```diff
@@ -50,26 +80,64 @@ The process of the hook files are provided with information about the event and
+Note that you must include `http://` before your url. 
 
-### post-finish
+### Usage
+Tusd will issue a `POST` request to the specified URL endpoint, specifying the hook name in the `Upload-State` header:
 
-This event will be triggered after an upload is fully finished, meaning that all chunks have been transfered and saved in the storage. After this point, no further modifications, except possible deletion, can be made to the upload entity and it may be desirable to use the file for further processing or notify other applications of the completions of this upload.
+Example: 
 
-### post-terminate
+Headers:
+```
+//Content Length of the JSON
+'Content-Length': '183', 
+'Accept-Encoding': 'gzip', 
+'User-Agent': 'Go-http-client/1.1', 
+//Specified host URL
+'Host': 'localhost:8081', 
+'Content-Type': 'application/json,
+//Hook that invoked this call. 
+'Upload-State': 'pre-create'
+```
 
-This event will be triggered after an upload has been terminated, meaning that the upload has been totally stopped and all associating chunks have been fully removed from the storage. Therefore, one is not able to retrieve the upload's content anymore and one may wish to notify further applications that this upload will never be resumed nor finished.
+Body: 
+```js
+{
+  // The upload's ID. Will be empty during the pre-create event
+  "ID": "14b1c4c77771671a8479bc0444bbc5ce",
+  // The upload's total size in bytes.
+  "Size": 46205,
+  // The upload's current offset in bytes.
+  "Offset": 1592,
+  // These properties will be set to true, if the upload as a final or partial
+  // one. See the Concatenation extension for details:
+  // http://tus.io/protocols/resumable-upload.html#concatenation
+  "IsFinal": false,
+  "IsPartial": false,
+  // If the upload is a final one, this value will be an array of upload IDs
+  // which are concatenated to produce the upload.
+  "PartialUploads": null,
+  // The upload's meta data which can be supplied by the clients as it wishes.
+  // All keys and values in this object will be strings.
+  // Be aware that it may contain maliciously crafted values and you must not
+  // trust it without escaping it first!
+  "MetaData": {
+    "filename": "transloadit.png"
+  }
+}
+```
 
-### post-receive
 
-This event will be triggered for every running upload to indicate its current progress. It will occur for each open PATCH request, every second. The offset property will be set to the number of bytes which have been transfered to the server, at the time in total. Please be aware that this number may be higher than the number of bytes which have been stored by the data store!
+### Configuration
+Tusd uses the [Pester library](https://github.com/sethgrid/pester) to issue requests and handle retries. By default, tusd will retry 3 times on a non-200 response, with a 1 second backoff. This can be configured with https://github.com/vimeo/tusd/blob/httphooks/cmd/tusd/cli/hooks.go#L103-L106.  
```

Comment:
Hmmmm, I'm not so sure about that, since they might have to add all sorts of weird stuff like auth headers/custom verification/custom backoff/etc. We might want to let them set basic things like back off and retry counter, but the reason I chose Pester is so that end-user can easily modify it.